sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -883,6 +883,31 @@ class Dataset[T] private[sql](
     join(right, Seq(usingColumn))
   }
 
+  /**
+    * Equi-join with another `DataFrame` using the given column.
+    *
+    * Different from other join functions, the join column will only appear once in the output,
+    * i.e. similar to SQL's `JOIN USING` syntax.
+    *
+    * {{{
+    *   // Left join of df1 and df2 using the column "user_id"
+    *   df1.join(df2, "user_id", "left")
+    * }}}
+    *
+    * @param right Right side of the join operation.
+    * @param usingColumn Name of the column to join on. This column must exist on both sides.
+    * @param joinType Type of join to perform. Default `inner`. Must be one of:
+    *                 `inner`, `cross`, `outer`, `full`, `full_outer`, `left`, `left_outer`,
+    *                 `right`, `right_outer`, `left_semi`, `left_anti`.
+    * @note If you perform a self-join using this function without aliasing the input
+    * `DataFrame`s, you will NOT be able to reference any columns after the join, since
+    * there is no way to disambiguate which side of the join you would like to reference.
+    * @group untypedrel
+    */
+  def join(right: Dataset[_], usingColumn: String, joinType: String): DataFrame = {
```

Comment:
What is the difference between this function and ```def join(right: Dataset[_], usingColumns: Seq[String], joinType: String)```?
https://github.com/apache/spark/blob/v2.4.0/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L932-L946

Reply:
So in your case. Could you replace ```df1.join(df2, "user_id", "left")``` with ```df1.join(df2, Seq("user_id"), "left")```?

Reply:
@arman1371 What do you think? ```def join(right: Dataset[_], usingColumn: String, joinType: String)``` only support one column. right?

Reply:
The answer of both questions are yes.

Reply:
Cloud we close this?

Reply:
no, it's a simpler implementation. as i said we have both of `def join(right: Dataset[_], usingColumn: String)` and `def join(right: Dataset[_], usingColumns: Seq[String])`. based on your opinion the first function should be removed.

Reply:
cc @dongjoon-hyun

Reply:
@arman1371 . We understand that this PR is trying to add a syntactic sugar. But you need only 5 characters, `Seq(` and `)`, to use the existing general API. Personally, I agree with @wangyum . I prefer not to add this.

Historically, 
1. Spark 1.4 adds `Seq[String]` version was added later to support PySpark (SPARK-7990)
2. Spark 1.6 adds `join` type to `Seq[String]` version (SPARK-10446)

It's a long time ago. Given that, I guess Apache Spark community intentionally didn't add the `String` version for this in order to keep `Dataset` simple in terms of the number of APIs. Anyway, since you need an answer, let's ask the general opinion again to make a decision.

Hi, @rxin, @cloud-fan , @gatorsmile  . Did we explicitly decide not to add this API ? It seems that @arman1371 wants to add this for feature parity with PySpark at Spark 3.0.0.

Reply:
Yes, for scala api it's just 5 characters but in java api it's very hard to change `String` to `Seq[String]`

Reply:
Normally, we do not add such an API.